core/arch/arm/kernel/boot.c

```diff
@@ -1385,6 +1390,10 @@ void init_tee_runtime(void)
 	 */
 	thread_init_core_local_pauth_keys();
 	thread_init_thread_pauth_keys();
+
+	/* Reinitialize canaries around the stacks with crypto_rng_read() */
```

Comment:
It would be nice to say why the canaries are not updated when virtualization is enabled.